### PR TITLE
Update Implementation.py

### PR DESCRIPTION
**Pull request by Karthikrock7**
Line 138 throws an attribute error as Nonetype object has no attribute data if we give a value not in list as an attribute. So, I have wrapped it in try block and excepted the attribute error and removed the else block. If there is a better solution, please share.

### CODE REVIEW

File: venv/Scripts/Data Structures/Linked Lists/Implementation.py [hunk 1]
```diff
@@ -135,20 +135,20 @@ def delete_by_value(self, data):
...
-            current_node = current_node.next
-        if current_node!=None:
-            current_node.next = current_node.next.next
-            if current_node.next == None:
-                self.tail = current_node
-            self.length -= 1
-            return
-        else:
+        try:
+            while current_node!= None and current_node.next.data != data:
```
[the remark this answers is not in the document]
Comment (Karthikrock7) [reply venv/Scripts/Data Structures/Linked Lists/Implementation.py:139]: Yeah.. This makes sense.

File: venv/Scripts/Data Structures/Linked Lists/Implementation.py [hunk 2]
```diff
@@ -135,20 +135,20 @@ def delete_by_value(self, data):
...
+                #    previous_node.next = current_node.next
+                #    return
+                current_node = current_node.next
+            if current_node!=None:
+                current_node.next = current_node.next.next
+                if current_node.next == None:
+                    self.tail = current_node
+                self.length -= 1
+                return
+        except AttributeError:
```
[the remark this answers is not in the document]
Comment (Karthikrock7) [reply venv/Scripts/Data Structures/Linked Lists/Implementation.py]: Yeah. By the changes you suggested, there will be no need for a try...except block.
If..else will do.

File: venv/Scripts/Data Structures/Linked Lists/Implementation.py [hunk 3]
```diff
@@ -135,20 +135,20 @@ def delete_by_value(self, data):
...
-            self.length -= 1
-            return
-        else:
+        try:
+            while current_node!= None and current_node.next.data != data:
+                #if current_node.data == data:
+                #    previous_node.next = current_node.next
+                #    return
+                current_node = current_node.next
+            if current_node!=None:
```
[the remark this answers is not in the document]
Comment (Karthikrock7) [reply venv/Scripts/Data Structures/Linked Lists/Implementation.py]: Ofcourse. This will wok

**Comment by Karthikrock7** [comment]
I have made the changes you have suggested.

**Review by VicodinAbuser — APPROVED** [review]
Thank You. Merging it now.

**Pull request merged**